pkg/common-controller/groupsnapshot_delete_test.go

```diff
@@ -0,0 +1,138 @@
+				"", classGold, "groupsnapcontent-snapuid1-1", &False, nil, nil, false, false, &timeNowMetav1,
+			),
+			expectedGroupSnapshots: newGroupSnapshotArray(
+				"snap-1-1", "snapuid1-1", map[string]string{
+					"app.kubernetes.io/name": "postgresql",
+				},
+				"", classGold, "groupsnapcontent-snapuid1-1", &False, nil, nil, false, false, &timeNowMetav1,
+			),
+			initialGroupContents:  nil,
+			expectedGroupContents: nil,
+			initialClaims: withClaimLabels(
+				newClaimCoupleArray("claim1-1", "pvc-uid6-1", "1Gi", "volume6-1", v1.ClaimBound, &classGold),
+				map[string]string{
+					"app.kubernetes.io/name": "postgresql",
+				}),
+			initialVolumes: newVolumeCoupleArray("volume6-1", "pv-uid6-1", "pv-handle6-1", "1Gi", "pvc-uid6-1", "claim1-1", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classGold),
+			errors:         noerrors,
+			test:           testSyncGroupSnapshot,
+			expectSuccess:  true,
+		},
+		{
+			name: "6-2 - dynamic group snapshot have been deleted, retention policy set to retain - set 'being deleted' annotation",
+			initialGroupSnapshots: withGroupSnapshotFinalizers(
+				newGroupSnapshotArray(
+					"snap-1-1", "snapuid1-1", map[string]string{
+						"app.kubernetes.io/name": "postgresql",
+					},
+					"", classGold, "groupsnapcontent-snapuid1-1", &False, nil, nil, false, false, &timeNowMetav1,
+				),
+				utils.VolumeGroupSnapshotBoundFinalizer,
+			),
+			expectedGroupSnapshots: newGroupSnapshotArray(
+				"snap-1-1", "snapuid1-1", map[string]string{
+					"app.kubernetes.io/name": "postgresql",
+				},
+				"", classGold, "groupsnapcontent-snapuid1-1", &False, nil, nil, false, false, &timeNowMetav1,
+			),
+			initialGroupContents: newGroupSnapshotContentArray(
+				"groupsnapcontent-snapuid1-1", "snapuid1-1", "snap-1-1", "snapshot-handle", classGold, []string{
+					"1-pv-handle6-1",
+					"2-pv-handle6-1",
+				}, "", crdv1.VolumeSnapshotContentRetain, nil, false, false,
+			),
+			expectedGroupContents: withGroupContentAnnotations(
+				newGroupSnapshotContentArray(
+					"groupsnapcontent-snapuid1-1", "snapuid1-1", "snap-1-1", "snapshot-handle", classGold, []string{
+						"1-pv-handle6-1",
+						"2-pv-handle6-1",
+					}, "", crdv1.VolumeSnapshotContentRetain, nil, false, false,
+				),
+				map[string]string{
+					utils.AnnVolumeGroupSnapshotBeingDeleted: "yes",
+				},
+			),
+			initialClaims: withClaimLabels(
+				newClaimCoupleArray("claim1-1", "pvc-uid6-1", "1Gi", "volume6-1", v1.ClaimBound, &classGold),
+				map[string]string{
+					"app.kubernetes.io/name": "postgresql",
+				}),
+			initialVolumes: newVolumeCoupleArray("volume6-1", "pv-uid6-1", "pv-handle6-1", "1Gi", "pvc-uid6-1", "claim1-1", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classGold),
+			errors:         noerrors,
+			test:           testSyncGroupSnapshot,
+			expectSuccess:  true,
+		},
+		{
+			name: "6-3 - dynamic group snapshot have been deleted, retention policy set to delete - volume snapshot content will be deleted",
```

Comment:
Add a test for pre-provisioned group snapshot?

Reply:
The test classification was not precise, I marked the pre-provisioned ones in a clearer way.